Mathlib/CategoryTheory/Comma/OverClass.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2024 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.Tactic.CategoryTheory.Reassoc
+import Mathlib.CategoryTheory.Comma.Over
+
+/-!
+# Typeclasses for `S`-objects and `S`-morphisms
+
+**Warning**: This is not usually how typeclasses should be used.
+This is only a sensible approach when the morphism is considered as a structure on `X`,
+typically in algebraic geometry.
```

Comment:
Please make the analogy with `[Algebra A B]` where a ring-hom is also considered as structure.

I think this file is doing a natural thing. But just like with `RingHom` vs `Algebra`, we will probably want to duplicate a lot of properties of morphisms into properties of `S`-objects.
So when the time comes, we should also generalize `algebraize`.

The reason I would like the analogy with `Algebra` to be made explicit, is that I think we can copy several design decisions from what we've learned there.

Reply:
By "make the analogy explicit" do you mean to mention it in the docstring or to add real code on `OverClass CommRing <-> Algebra`

Reply:
I meant the docstring.